src/Rules/Wrappers.hs

```diff
@@ -49,15 +51,16 @@ installRunGhcWrapper WrappedBinary{..} = do
 inplaceGhcPkgWrapper :: WrappedBinary -> Expr String
 inplaceGhcPkgWrapper WrappedBinary{..} = do
     expr $ need [sourcePath -/- "Rules/Wrappers.hs"]
-    top <- expr topDirectory
+    stage <- succ <$> getStage
```

Comment:
Let's rename this to `nextStage` to avoid confusion.

Reply:
I have not tested the wrappers rules yet. They're less necessary with relocatable ghcs -- also one of the topics I brought up in the email IIRC, that I wanted to get your opinion on. Will make this change but would appreciate your opinion on the wrappers in general.

Reply:
I am quite hazy on wrappers, because I use Windows and they are not built on Windows (I guess exactly because Windows already had some relocatable GHCs). I think it would be awesome if we could get rid of wrappers for good.

Reply:
We will need wrapper for anything but win, lin and mac. Those are the only platform for which we have somewhat sane support of detecting the executables path. BSD might work as well. Anything else will likely still require wrappers.

I'd argue we should drop wrappers altogether from hadrian. The improved distrib configure script @alpmestan built can generate wrappers at install time, which would leave us with wrappers only missing at development time for exotic architectures.

Reply:
To be fair, [my bindist rule](https://github.com/alpmestan/hadrian/commit/b6d197e3f9a1c13c1bfd4154a9cd61b7178236b4) only generates the GHC binary wrapper for now. But I told @chitrak7 all about it and what's missing to "compete" with the make build system's bindists, so hopefully we'll have something much smarter and complete.

Reply:
@angerman Sounds good with me. I hope there are not many GHC developers who are doing GHC development on exotic platforms.